server/src/main/java/org/elasticsearch/index/mapper/vectors/SparseVectorFieldMapper.java

```diff
@@ -161,6 +156,7 @@ public void parse(DocumentParserContext context) throws IOException {
         }
 
         String feature = null;
+        float value = 0;
```

Comment:
we shouldn't do it this way. Instead something like `bool wroteField;` is way clearer and less prone to bugs. 

What if we support `0` weighted dimensions in the future?

Reply:
Tbh having clear and well defined (scope and usage) variables is my preference as well, I just couldn't envision a scenario where a weight of `0` would be of any value in this specific case. But I guess that it's just as likely that we may very well  support negative values sometime in the future, so the above (as expected) is error-prone 😞 

Will update accordingly!  :)